src/core/validations/message.rs

```diff
@@ -64,6 +67,22 @@ fn validate_frame_action_body(body: &FrameActionBody) -> Result<(), ValidationEr
     Ok(())
 }
 
+pub fn validate_timestamp(
+    message_timestamp: u32,
+    engine_version: EngineVersion,
+) -> Result<(), ValidationError> {
+    if engine_version.is_enabled(ProtocolFeature::FutureTimestampValidation) {
+        let current_time = get_farcaster_time()? as u32;
```

Comment:
Should this use the block timestamp?

Reply:
Yeah actually I think that's better so that the check is deterministic.